osu.Game/Screens/Edit/Timing/TimingScreen.cs

```diff
@@ -149,16 +152,27 @@ protected override void Update()
             private void trackActivePoint()
             {
                 // For simplicity only match on the first type of the active control point.
-                var selectedPointType = selectedGroup.Value?.ControlPoints.FirstOrDefault()?.GetType();
+                if (selectedGroup.Value == null)
+                    trackedType = null;
+                else
+                {
+                    // If the selected group only has one control point, update the tracking type.
+                    if (selectedGroup.Value.ControlPoints.Count == 1)
+                        trackedType = selectedGroup.Value?.ControlPoints.Single().GetType();
+                    // If the selected group has more than one control point, choose the first as the tracking type
+                    // if we don't already have a singular tracked type.
+                    else if (trackedType == null)
+                        trackedType = selectedGroup.Value?.ControlPoints.FirstOrDefault()?.GetType();
```

Comment:
Side note - currently are only two possible control point types on this screen, and therefore, if there is more than one control point in the group, then all applicable types of control points are in the group.

If there are ever more, then this logic may lead to unexpected results. Imagine selecting a control point group with types `[A, B]`, followed by selecting a group with types `[C, D]`. The first selection would cause the timing screen to start tracking `A`, but the second would not change tracking at all to either `C` or `D` (because there is more than one type in the group, and the tracked type is already set).

Not really worth fixing, just something that occurred in my mind in passing.

Reply:
Yeah, it's still best-effort. If we add more, I'd want the current selected "type" to show somewhere in the interface, and tracking be based on which you interacted with.